deploy/maven-repo.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: jvm-build-maven-repo
+  namespace: test-jvm-namespace
+spec:
+  revisionHistoryLimit: 0
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: jvm-build-maven-repo
+  template:
+    metadata:
+      labels:
+        app: jvm-build-maven-repo
+    spec:
+      serviceAccountName: pipeline
+      securityContext:
+        runAsUser: 0
+      containers:
+        - name: jvm-build-maven-repo
+          image: dzikoysk/reposilite:3.5.14
```

Comment:
I wonder if this should be `dzikoysk/reposilite:latest` so we pick up eg 3.5.16. Equally I guess that might introduce some instability risk. The alternative would be to use a scheme like ./pkg/reconciler/dependencybuild/scripts/Dockerfile.build-trusted-artifacts and have a Dockerfile.reposilite and substitute in via a variable. This image would then be updated by renovate.

Reply:
It could potentially introduce some instability risk pinning 'latest' version that's true. Do we actually need to keep the image up to date given that we only use it for testing?

Reply:
Thats a very good question! In general it probably shouldn't matter however given 3.5.12 (for example) had a number of vulnerabilities I'm thinking we should at least have a way of it being updated to avoid being flagged for using vulnerable dependencies. However, I'm ok if this is moved to a separate piece of work (as both the above reference and the one in the test suite would need to be handled) if you want (and a ticket is entered)

Reply:
I believe I've made the necessary changes for renovate to update reposilite to latest version. Please let me know if anything is messing!

Reply:
That looks great!